src/util/configure.ml

```diff
@@ -30,6 +32,17 @@ let main () =
   print"use_pthread" !use_pthread;
   print"android_target" !android_target;
   print"libev_default" !libev_default;
+  close_out f;
+  (* '-use-camlp4 false' will write a jbuild-ignore file directing jbuilder to ignore
+   *                     the camlp4 directory. 
+   * '-use-camlp4 true' (or none) will write an empty file which does nothing 
+   *                     except to potentially overwrite any previous version. 
+   * 
+   * This is a workaround required to overcome some weird camlp4 packaging behaviour
+   * where ocamlfind sometimes installs a dummy META file for it even though it's 
+   * not actually installed. *)
+  let f = open_out "src/jbuild-ignore" in
+  (if !use_camlp4 = Some(false) then Printf.fprintf f "camlp4" else ());
```

Comment:
I think `camlp4` should be off by default; that might also help with the overall review comment.